The Lua interpreter doesn't have a hard dependency on readline so build it
without it. This means that we don't need to make readline available in all
environments where we build the interpreter (it seems to not be available on
Ubuntu since at least 24).

--- a/src/luaconf.h	2008-02-12 00:25:08.000000000 +0800
+++ b/src/luaconf.h	2025-04-13 18:57:35.608033074 +0800
@@ -270,6 +270,7 @@
 ** CHANGE them if you want to improve this functionality (e.g., by using
 ** GNU readline and history facilities).
 */
+#undef LUA_USE_READLINE
 #if defined(LUA_USE_READLINE)
 #include <stdio.h>
 #include <readline/readline.h>

--- a/src/Makefile	2012-02-14 04:41:22.000000000 +0800
+++ b/src/Makefile	2025-04-13 18:59:08.835533473 +0800
@@ -96,10 +96,10 @@
 	$(MAKE) all MYCFLAGS=

 linux:
-	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS="-Wl,-E -ldl -lreadline -lhistory -lncurses"
+	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS="-Wl,-E -ldl -lncurses"

 macosx:
-	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS="-lreadline"
+	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS=""
 # use this on Mac OS X 10.3-
 #	$(MAKE) all MYCFLAGS=-DLUA_USE_MACOSX
